stdlib/builtin/complex.rbs

```diff
@@ -1,400 +1,367 @@
 #     '1@2'.to_c           #=> (-0.4161468365471424+0.9092974268256817i)
-# 
+#
 # A complex object is either an exact or an inexact number.
-# 
+#
 #     Complex(1, 1) / 2    #=> ((1/2)+(1/2)*i)
 #     Complex(1, 1) / 2.0  #=> (0.5+0.5i)
-# 
+#
 class Complex < Numeric
+  public
+
   # Returns a complex object which denotes the given polar form.
-  # 
+  #
   #     Complex.polar(3, 0)            #=> (3.0+0.0i)
   #     Complex.polar(3, Math::PI/2)   #=> (1.836909530733566e-16+3.0i)
   #     Complex.polar(3, Math::PI)     #=> (-3.0+3.673819061467132e-16i)
   #     Complex.polar(3, -Math::PI/2)  #=> (1.836909530733566e-16-3.0i)
-  # 
+  #
   def self.polar: (Numeric, ?Numeric) -> Complex
 
   # Returns a complex object which denotes the given rectangular form.
-  # 
+  #
   #     Complex.rectangular(1, 2)  #=> (1+2i)
-  # 
+  #
   def self.rect: (Numeric, ?Numeric) -> Complex
 
-  # Returns a complex object which denotes the given rectangular form.
-  # 
-  #     Complex.rectangular(1, 2)  #=> (1+2i)
-  # 
   alias self.rectangular self.rect
 
-  public
-
   # Performs multiplication.
-  # 
+  #
   #     Complex(2, 3)  * Complex(2, 3)   #=> (-5+12i)
   #     Complex(900)   * Complex(1)      #=> (900+0i)
   #     Complex(-2, 9) * Complex(-9, 2)  #=> (0-85i)
   #     Complex(9, 8)  * 4               #=> (36+32i)
   #     Complex(20, 9) * 9.8             #=> (196.0+88.2i)
-  # 
+  #
   def *: (Numeric) -> Complex
 
   # Performs exponentiation.
-  # 
+  #
   #     Complex('i') ** 2              #=> (-1+0i)
   #     Complex(-8) ** Rational(1, 3)  #=> (1.0000000000000002+1.7320508075688772i)
-  # 
+  #
   def **: (Numeric) -> Complex
 
   # Performs addition.
-  # 
+  #
   #     Complex(2, 3)  + Complex(2, 3)   #=> (4+6i)
   #     Complex(900)   + Complex(1)      #=> (901+0i)
   #     Complex(-2, 9) + Complex(-9, 2)  #=> (-11+11i)
   #     Complex(9, 8)  + 4               #=> (13+8i)
   #     Complex(20, 9) + 9.8             #=> (29.8+9i)
-  # 
+  #
   def +: (Numeric) -> Complex
 
   def +@: () -> Complex
 
   # Performs subtraction.
-  # 
+  #
   #     Complex(2, 3)  - Complex(2, 3)   #=> (0+0i)
   #     Complex(900)   - Complex(1)      #=> (899+0i)
   #     Complex(-2, 9) - Complex(-9, 2)  #=> (7+7i)
   #     Complex(9, 8)  - 4               #=> (5+8i)
   #     Complex(20, 9) - 9.8             #=> (10.2+9i)
-  # 
+  #
   def -: (Numeric) -> Complex
 
   # Returns negation of the value.
-  # 
+  #
   #     -Complex(1, 2)  #=> (-1-2i)
-  # 
+  #
   def -@: () -> Complex
 
   # Performs division.
-  # 
+  #
   #     Complex(2, 3)  / Complex(2, 3)   #=> ((1/1)+(0/1)*i)
   #     Complex(900)   / Complex(1)      #=> ((900/1)+(0/1)*i)
   #     Complex(-2, 9) / Complex(-9, 2)  #=> ((36/85)-(77/85)*i)
   #     Complex(9, 8)  / 4               #=> ((9/4)+(2/1)*i)
   #     Complex(20, 9) / 9.8             #=> (2.0408163265306123+0.9183673469387754i)
-  # 
+  #
   def /: (Numeric) -> Complex
 
-  def <: (Numeric) -> bot
```

Comment:
I think the `-> bot` methods should not be removed. Because of inheritance, `<` method will be defined, and RBS does not have any way to remove an inherited method. So, returning `bot` means we cannot invoke the method (== it implies it raises an error).

Having `-> bot` methods here is intentional. (If you believe removing them makes more sense, feel free to explain your thoughts. 😄 )

Reply:
I do not understand on `bot`. I try to retrieve all change on `bot`.